addons/mrp/tests/test_order.py

```diff
@@ -4916,6 +4916,80 @@ def test_mo_without_resource_calendar(self):
         self.assertEqual(mo.state, 'done')
         self.assertEqual(mo.workorder_ids[0].duration_expected, 1440.0)
 
+    def test_additional_transfer_creation_in_progress_state(self):
+        """
+        Test the creation of additional component transfers for MOs in 'progress' and 'to_close'
+        state for both 2-step and 3-step MRP processes.
```

Comment:
It would be also intersting to test when the picking is not validated + check the quantity inside